dali/pipeline/executor/executor.h

```diff
@@ -537,6 +537,8 @@ void Executor<WorkspacePolicy, QueuePolicy>::PruneUnusedGraphNodes() {
       OpNode &node = graph_->Node(i);
       // If this node has children, don't prune it
       if (!node.children.empty()) continue;
+      // Do not prune the node if it explicitly has no outputs
+      if (node.spec.NumOutput() == 0) continue;
```

Comment:
I would prefer to have a `spec.IsSink()` member function that would by default check the number of outputs but would also allow to be set by some schema parameter.

I am not sure if we want to support scenarios like: sink operator that produces output, Ops after that sink are removed from graph, but the sink stays.

Reply:
I think we may have something that is a sink and still produces an output. The idea behind the sink is to make this operator stay in the graph no mater if the output is used or not.

Reply:
Maybe: add a flag .allowSideEffects() to an operator spec and a function spec.canBePruned() which would check if an operator has the side effects flag OR has zero outputs.

Reply:
Just a boolean in schema to set the operator as sink?
We discussed before having an member function in Operator Base and overloading it for those that can be sinks, but it is conflicting with how we could generate a docs (I think we only do it based on schema).